src/searcher/glue.rs

```diff
@@ -132,7 +117,7 @@ impl<'s, S: Sink> MultiLine<'s, S> {
 
     fn sink<'tree>(
         &mut self,
-        matches: &mut impl Iterator<Item = Node<'tree>>,
+        matches: &mut impl StreamingIterator<Item = CaptureInfo<'tree>>,
```

Comment:
That DRY'ed up `get_captures()` returns a `StreamingIterator` which has nothing to do with async "streams", it's a crate for doing a "lending" iterator (ie where the items returned are dependent on the iterator's lifetime)